duckduckgo/Android
# Ignore applinks for current domain

Task/Issue URL: https://app.asana.com/0/414730916066338/1200868385373197/f

### Description
This PR adds logic to ignore app links for the current domain if the user taps "no" on the app link dialog.

### Steps to test this PR

Install a companion app (Amazon, LinkedIn, Github, Guardian, Wikipedia...)
The below steps assume the Amazon app was installed.

_Navigating via SERP_
1. Type "amazon" in the search bar
2. Tap on the Amazon link in the search results
3. Verify that the app link dialog is shown
4. Tap "No" on the dialog
5. Verify that the app link dialog is not shown again while navigating around the amazon.com domain
6. Type "google maps" in the search bar
7. Tap on the Google Maps link in the search results
8. Verify that the app link dialog is shown

_Navigating via search bar_
1. Type "amazon.com" in the search bar
2. Verify that the app link dialog is not shown while navigating around the amazon.com domain
